src/main/scala/com/databricks/labs/mosaic/models/util/CheckpointManager.scala

```diff
@@ -0,0 +1,105 @@
+package com.databricks.labs.mosaic.models.util
+
+import org.apache.spark.sql.Dataset
+
+/**
+  * CheckpointManager is a utility class to manage checkpointing of the mosaic
+  * model.
+  *
+  * @param checkpoint
+  *   Either DeltaFile or DeltaTable checkpoint instance.
+  */
+case class CheckpointManager(checkpoint: DeltaCheckpoint) {
```

Comment:
Why do we need a dedicated class for checkpoointing? What is the difference with Spark checkpoint/localCheckpoint?